### ISSUE TITLE
[Feature]: Make night vision a passive for better vision in the dark

### ISSUE THREAD

**Issue by MarioSMB** [TeamLapen/Vampirism#1243]
### What?

It's always bugged me that the first skill you unlock as a vampire is... a potion effect, and one that makes the overworld ugly leading to one just turning it on to cheese the underground.  

In that vein, I'd like to suggest the following alterations to the skill:  
Make it a passive ability with no toggles and instead have the skill passively make caves brighter - effectively just a boost to the overall brightness of the game without making everything fullbright.

### Mod experience

It would add some uniqueness to the skill rather than taking the white bread of modding approach and copying vanilla potion effects; vampires would no longer have a need to toggle it as its effects would always be beneficial whilst not making potions redundant!  
"As a growing vampire your eyes slowly adjust to the darkness you now call home"

**Comment by lunofe** [comment]
Hm, I definitely see the point that maybe the whole thing should have some sort of automaticity and rather be a brightness boost instead of blistering fullbright.

But it doesn't make sense to limit this to caves only. Outside it is also dark at night, and it would be illogical not to have any night vision there. Lore-wise, vampires have to get blood somehow, and they certainly don't do that in caves.

Furthermore, there are certainly players who find it very good to turn night vision on or off as needed - for example, all those who play with shaders, because night vision causes funny bugs there.

**Comment by MarioSMB** [comment]
Indeed, it wouldn't only apply in caves, the effect would just be noticeable in dark areas.  
For those with shaders, I figure a client configuration option would be good enough to toggle the effect (possibly along with an option to control how much brighter it makes everything).

**Comment by lunofe** [comment]
Ah okay, then I had misunderstood you. If the minimum brightness value is increased in general, that sounds great. But I would still leave the existing hotkey as it is. Always looking into a config file is just too impractical, especially when you change shaders or turn them off in between.

**Comment by MarioSMB** [comment]
That is fair - I don't know exactly the use case of toggling shaders on and off mid gameplay but as long as the keybind is available it may as well be used.  
Though in my humble opinion having the keybind *only* be for blood vision would be nice so it isn't an awkward three-way toggle.

**Comment by lusifer173** [comment]
I would also like to mention as someone who likes to build, lighting is extremely important, even a slight change to what I think your saying is the lighting setting can really throw off the whole vibe of a build.  Also whats wrong with a potion effect?

**Comment by MarioSMB** [comment]
Potion effects are the white bread of "bonuses" in modding; every mod will have some special ability or item that permanently gives you the vanilla potion effects. I would hope Vampirism can go beyond that and offer a **unique** twist on gameplay, not just copying what vanilla already has.  

As for lighting - that is a fair criticism if the effect were to not have a toggle, though I don't know the exact logistics of building around a specific brightness setting given it can be adjusted so easily in the menu (and by default with some performance mods like Rubidium Extras).  
Something to consider for sure!

**Comment by lusifer173** [comment]
I think I'm confused on something, hopefully you can shed some light on this, but how is it a "unique twist" or "not just copying what vanilla already has" if its literally just copying the brightness setting. Unless im getting something wrong? This doesn't seem unique in anyway, also unless the twist is "hah we tricked you into thinking it was actually night vision" its not a twist either since its just a worse version, honestly I wouldn't even call it night vision at that point.

**Comment by MarioSMB** [comment]
I imagine it would allow dark areas to be rendered a bit brighter than the maximum brightness the slider goes to in the menu to give it some value.  
Indeed, it is very different to what a night vision potion would give and may be deserving of a new name if this change were to be considered.  

I understand that many players appreciate the ease of access these skills provide to effects that normally take potions and beacons, but I don't personally believe it makes for a unique or fun experience - not in the way genuinely good abilities can.   
Maybe seeing a bit better in the dark isn't what some would consider fitting of that clause but it is *something*; something I can't replicate in the vanilla game, something that helps being a vampire stand out and something that warrants the drawbacks that come with it. And most importantly, it's something I wouldn't want turned off most of the time.

**Comment by Cheaterpaul** [comment]
The think about modding is that we may change anything, but that means that any other mod may do the same, which then becomes an incompatibility. If Forge or Vanilla offer a way to do such things we will gladly use it, but if not we either need to change the behavior in vanilla, which is always a bad idea, or find a way to make this happen by using vanilla / forge feature.

Especially any changes to the rendering, which are used every frame, should be reasonable given their runtime, complexity and lack of compatibility. So most times it is better to stick with vanilla mechanics.

Unfortunately for this change of night vision there is no option for us to increase the brightness only for specific blocks. This is only possible for a general light level which the night vision changes. Other solutions are so complicated that we should write a mod for it.

**Comment by MarioSMB** [comment]
Boosting the general light level is pretty much what I'm suggesting; it doesn't need to be anything too fancy, just something that doesn't turn the world fullbright!

**Comment by maxanier** [comment]
> Potion effects are the white bread of "bonuses" in modding; every mod will have some special ability or item that permanently gives you the vanilla potion effects. I would hope Vampirism can go beyond that and offer a unique twist on gameplay, not just copying what vanilla already has.

I agree, but I believe Vampirism does so in various aspects and I don't think just re-using what vanilla has to offer is always bad. Why reinvent the wheel, when there already is something perfect available. 
Also, as @lusifer173 pointed out, just increasing the maximum brightness slider limit is not something overly creative.

**Comment by MarioSMB** [comment]
The problem with this wheel in particular is that you can't really play with night vision on most of the time as it makes the world really ugly.  
A flat brightness boost isn't overly unique either, but vampires *do* need some way to see better in the dark and it would ideally be something that can be left on and/or isn't an inconvenience needing toggling.

**Comment by kaboom24** [comment]
> The problem with this wheel in particular is that you can't really play with night vision on most of the time as it makes the world really ugly.<

That's more of an opinion,  I think it looks quite fine!

**Comment by MarioSMB** [comment]
Making night vision a toggle is the happy medium that lets everyone have it how they want... except it adds a level of nuisance to playing as a vampire; you either love fullbright and keep it on 24/7 or you're constantly needing to turn it on and off whenever you enter dark spaces (or of course you leave it off and gain nothing).  

Terraria's version of night vision would be perfect for vampirism, but understandably as mentioned previously it would not be easy to implement in Minecraft.  

Devil's advocacy would dictate that no action should be taken, since there are those who want it one way and those who want it the other, and the current configuration allows both.  
But it can't hurt to consider alternatives, at the frightening cost of change!
